DataFormats/ForwardDetId/interface/HGCalDetId.h

```diff
@@ -59,6 +59,27 @@ class HGCalDetId : public DetId {
   static const HGCalDetId Undefined;
 };
 
+namespace HGCalDetIdUtil
+{
+  // Functions to work on raw HGCAlDetIds
+  int subdetIdOf(uint32_t id);
+  int cellOf(uint32_t id);
+  int waferOf(uint32_t id);
+  int waferTypeOf(uint32_t id);
+  int layerOf(uint32_t id);
+  int zsideOf(uint32_t id);
+
+  void setCellOf(uint32_t& id, int cell);
+  void setWaferOf(uint32_t& id, int mod);
```

Comment:
And, consequently have this function return the new detid, or simply forcibly inline this function.